vbfml/scripts/err_analysis

```diff
@@ -455,5 +489,313 @@ def compare_hist(ctx, input_dirs: str, number_epochs: int, save_name: str) -> No
     plt.savefig(pjoin(outdir, f"comparison_val_loss_{save_name}.pdf"))
 
 
+@cli.command()
+@click.pass_context
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="path to the prediction directory of a model",
+)
+def roc(ctx, input_dir: str) -> None:
+    """
+    compute the roc curve of the model. /!/ input_dir point at the prediction_dir without a '/' at the end.
+    only compute and write in pkl file so it allows to use plot-roc on several model to compare in in one plot
+    """
+
+    # load model and prediction pickle file
+    loader = TrainingLoader(os.path.dirname(input_dir))
+    model = loader.get_model()
+    features, predictions, label_encoding, labels = load_cache(input_dir)
+    mjj = features["mjj"]
+    detajj = features["detajj"]
+
+    # compute ROC and precision recall curve
+    precision, recall, thresholds_prc = precision_recall_curve(labels, predictions)
+    fpr, tpr, thresholds_roc = roc_curve(labels, predictions)
+
+    # do the same for a simple cut on mjj and detajj
+    mjj_norm = mjj / mjj.max()
+    detajj_norm = detajj / detajj.max()
+    fpr_m, tpr_m, thresh_roc_m = roc_curve(labels, mjj_norm)
+    fpr_eta, tpr_eta, thresh_roc_eta = roc_curve(labels, detajj_norm)
+
+    # write data into pickle file
+    print("write data into pickle file")
```

Comment:
It's nice that you're logging what the function is doing. To improve this a little, can we do something like:

```
pkl_output_file = pjoin(input_dir, "roc.pkl")
# Mention the output file path in the logging
print(f"Writing output to: {pkl_output_file}")

# And then later...
with open(pkl_output_file, "wb") as f:
    pickle.dump(data, f)

```

Reply:
done